virttest/utils_test/qemu.py

```diff
@@ -1842,3 +1843,301 @@ def get_all_memorys(self, vm):
         session = self.get_session(vm)
         output = session.cmd_output_safe(cmd, timeout=90)
         return set([_ for _ in output.splitlines() if _])
+
+
+class UsbDevTest(object):
+
+    """
+    Provide basic test for usb devices
+    """
+
+    def __init__(self, test, params, env):
+        self.params = params
+        self.test = test
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+
+    def check_usb_dev(self, usb_type, info_usb_name,
+                      vp_id, vendor, product):
+
+        """
+        check if the usb device exist in guest
+        """
+        if usb_type in ['usb-storage', 'usb_bot']:
+            self.add_udisk()
+        timeout = float(self.params.get("login_timeout", 600))
+        self.vm.verify_alive()
+        session = self.vm.wait_for_login(timeout=timeout)
+        logging.info("Verify USB device in monitor.")
+        o = self.vm.monitor.info("usb")
+        if isinstance(o, dict):
+            o = o.get("return")
+        if info_usb_name and (info_usb_name not in o):
+            raise exceptions.TestFail("Could not find '%s' device, monitor "
+                                      "returns: \n%s" % (usb_type, o))
+        logging.info("The %s device can find in the monitor." % usb_type)
+        exist = self.check_usb_dev_guest(usb_type, info_usb_name, session,
+                                         vp_id, vendor, product)
+        if exist:
+            logging.info("The  %s device was found in guest" % usb_type)
+        else:
+            raise exceptions.TestFail("Could not find '%s' in guest" %
+                                      usb_type)
+
+    def check_usb_dev_guest(self, usb_type, info_usb_name,
+                            session, vp_id, vendor, product):
+        chk_list = []
+        if vp_id:
+            chk_list.append(vp_id)
+        if vendor:
+            chk_list.append(vendor)
+        if product:
+            chk_list.append(product)
+        logging.info("Verify %s device in guest." % usb_type)
+        for item in chk_list:
+            output = session.cmd_output("lsusb -v")
+            print output
+            devices = re.findall(item, output, re.I | re.M)
+            if not devices:
```

Comment:
sometimes,there is no output of idProduct using just "lsusb",so I need to get the parameter from iproduct.yes,some usb device have the same  parameter,but one case only test one usb device ,so I think it will be ok.